### PR TITLE
hack/verify-description.sh: correctly look for versioned API types

### PR DESCRIPTION
**Pull request by nikhita**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:

`find_files` had incorrect regexes which led to `genswaggertypedocs`
never being invoked. This commit updates the regex to correctly
look for versioned API types.go files.

Since `genswaggertypedocs` was never invoked, there are many API types
with missing descriptions. This commit lists these `types.go` files
in `hack/.descriptions_failures`.

The `verify-description.sh` script skips running `genswaggertypedocs`
on the files listed in `hack/.descriptions_failures`.

#### Special notes for your reviewer:

Noticed it while working on https://github.com/kubernetes/kubernetes/pull/99307.

This script hasn't been working for a  _long_ time. It used to look at files like `pkg/api/v1beta1/types.go` (ref: https://github.com/kubernetes/kubernetes/pull/2482#issuecomment-63883953) because there was no staging back then. It was not adapted to look at the APIs in staging when the pkg -> staging move happened (https://github.com/kubernetes/kubernetes/pull/44784).

If this looks good to be merged, I can also create a good first issue to add descriptions for the `types.go` files listed in `hack/.descriptions_failures`.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

/assign @dims

### CODE REVIEW

**Comment by nikhita** [comment]
Probably:
/sig architecture
/area code-organization

/priority backlog

File: hack/verify-description.sh [hunk 1]
```diff
@@ -45,25 +45,38 @@ find_files() {
...
+
 for file in $versioned_api_files; do
-  $genswaggertypedocs -v -s "${file}" -f - || gen_swagger_result=$?
-  if [[ "${gen_swagger_result}" -ne "0" ]]; then
-    echo "API file: ${file} is missing: ${gen_swagger_result} descriptions"
-    result=1
+  if ! kube::util::array_contains "$file" "${failing_files[@]}"; then
+    gen_swagger_result=0
+    $genswaggertypedocs -v -s "${file}" -f - || gen_swagger_result=$?
+    if [[ "${gen_swagger_result}" -ne "0" ]]; then
```
Comment (joakimr-axis) [inline hack/verify-description.sh:74]: Drop quotes for numeric (as clearly indicated by `-ne` value 0.

`if [[ "${gen_swagger_result}" -ne 0 ]]; then`

Comment (nikhita) [reply hack/verify-description.sh]: Thanks, done!

**Comment by dims** [comment]
/approve
/assign @liggitt 

thanks for taking care of this. will leave `/lgtm` to @liggitt

File: hack/verify-description.sh [hunk 2]
```diff
@@ -45,25 +45,38 @@ find_files() {
         -o -wholename './target' \
         -o -wholename '*/third_party/*' \
         -o -wholename '*/vendor/*' \
+        -o -wholename './pkg/*' \
       \) -prune \
     \) \
-    \( -wholename '*pkg/apis/*/v*/types.go' \
-       -o -wholename '*pkg/api/unversioned/types.go' \
-    \)
+    -wholename './staging/src/k8s.io/api/*/v*/types.go'
```
Comment (liggitt) [inline hack/verify-description.sh:51]: we also have API definitions in `staging/src/k8s.io/kube-aggregator/pkg/apis/*/v*/types.go` and `staging/src/k8s.io/apiextensions-apiserver/pkg/apis/*/v*/types.go`

**Comment by liggitt** [comment]
thanks for fixing this up. noted two more staging repos containing API types, then lgtm

File: hack/verify-description.sh (hunk 2 above)
Comment (nikhita) [reply hack/verify-description.sh]: done :+1:

**Comment by nikhita** [comment]
I will create a good first issue later to update the types.go files in `hack/.description_failures`.

**Pull request merged**